core/domain/suggestion_services_test.py

```diff
@@ -5998,3 +5999,234 @@ def test_create_email_task_raises_exception_for_wrong_contribution_subtype(
+            'skill_1', 1,
+            suggestion_models.STATUS_IN_REVIEW, self.author_id,
+            'reviewer_2', suggestion_change, 'category1',
+            'thread_1', 'en')
+        username = user_services.get_username(self.author_id)
+        from_date = datetime.datetime.today() - datetime.timedelta(days=1)
+        to_date = datetime.datetime.today() + datetime.timedelta(days=1)
+
+        response = suggestion_services.generate_contributor_certificate(
+            username,
+            feconf.SUGGESTION_TYPE_ADD_QUESTION,
+            None,
+            from_date,
+            to_date
+        )
+
+        self.assertIsNotNone(response)
+        # Generated image file of the certificate is deleted.
+        os.remove(response)
+
+    def test_create_contributor_certificate_raises_exception_for_no_suggestions(
+        self
+    ) -> None:
+        username = user_services.get_username(self.author_id)
+        from_date = datetime.datetime.today() - datetime.timedelta(days=1)
+        to_date = datetime.datetime.today() + datetime.timedelta(days=1)
+
+        with self.assertRaisesRegex(
+            Exception,
+            'There are no contributions for the given time range.'
+        ):
+            suggestion_services.generate_contributor_certificate(
+                username,
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                'hi',
+                from_date,
+                to_date
+            )
+
+    def test_create_certificate_raises_exception_for_no_question_suggestions(
+        self
+    ) -> None:
+        username = user_services.get_username(self.author_id)
+        from_date = datetime.datetime.today() - datetime.timedelta(days=1)
+        to_date = datetime.datetime.today() + datetime.timedelta(days=1)
+
+        with self.assertRaisesRegex(
+            Exception,
+            'There are no contributions for the given time range.'
+        ):
+            suggestion_services.generate_contributor_certificate(
+                username,
+                feconf.SUGGESTION_TYPE_ADD_QUESTION,
+                None,
+                from_date,
+                to_date
+            )
+
+    def test_create_contributor_certificate_raises_exception_for_wrong_language(
+        self
+    ) -> None:
+        username = user_services.get_username(self.author_id)
+        from_date = datetime.datetime.today() - datetime.timedelta(days=1)
+        to_date = datetime.datetime.today() + datetime.timedelta(days=1)
+
+        with self.assertRaisesRegex(
+            Exception,
+            'The language that is provided is invalid.'
+        ):
+            suggestion_services.generate_contributor_certificate(
+                username,
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                'test',
+                from_date,
+                to_date
+            )
+
+    def test_create_contributor_certificate_raises_exception_for_wrong_type(
+        self) -> None:
```

Comment:
```suggestion
    def test_create_contributor_certificate_raises_exception_for_wrong_type(
        self
    ) -> None:
```

ditto elsewhere

I've already mentioned it in this PR multiple times.

Reply:
Done